File: android/CriminalIntent/app/src/main/java/com/bignerdranch/android/criminalintent/SingleFragmentActivity.java
package com.bignerdranch.android.criminalintent;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;

public abstract class SingleFragmentActivity extends FragmentActivity {

    protected abstract Fragment createFragment();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        //Inflate the activity's view
        setContentView(R.layout.activity_fragment);

        //Get the fragment manager
        FragmentManager fm = getSupportFragmentManager();

        //Add a crime fragment
        Fragment fragment = fm.findFragmentById(R.id.fragment_container);

        //Create a new fragment transaction, add and commit it
        if(fragment == null){
            fragment = createFragment();
            fm.beginTransaction()
                    .add(R.id.fragment_container, fragment)
                    .commit();
        }

    }

}

File: android/CriminalIntent-Kotlin/app/src/main/java/com/me/android/criminalintent/SingleFragmentActivity.kt
package com.me.android.criminalintent

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.appcompat.app.AppCompatActivity

abstract class SingleFragmentActivity: AppCompatActivity() {
    protected abstract fun createFragment(): Fragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_fragment)

        var fragment: Fragment? = supportFragmentManager.findFragmentById(R.id.fragment_container)
        if (fragment == null) {
            fragment = createFragment()
            supportFragmentManager.beginTransaction()
                .add(R.id.fragment_container, fragment)
                .commit()
        }

    }
}